packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts

```diff
@@ -28,7 +28,7 @@ type Constructor = (...args: Parameters<typeof defaultDayjs>) => Dayjs;
 
 const formatTokenMap: FieldFormatTokenMap = {
   // Year
-  YY: 'year',
+  YY: { sectionType: 'year', contentType: 'digit', maxLength: 2 },
```

Comment:
Could be related, but I noticed on vitest an error that could only be fixed by adding

`yy: { sectionType: 'year', contentType: 'digit', maxLength: 2 },` to `AdapterDateFnsBase`.

I don't understand why this is the case or why the tests were failing only in vitest, so if you create a fix let me know, otherwise I'll isolate the change into a PR later.

Reply:
I'm not sure why it was not failing before :laughing: 
`YY` returns `1`, `2`, ... `99`, so we need `maxLength` on it.
But I have a small issue with keyboard editing and `YY` that I'm trying to fix (if I can), I'll merge after :+1:

Reply:
```
console.log(dayjs().set('year', 2009).format('YY')); // 09
console.log(dayjs().set('year', 9).format('YY')); // 9
```

I don't want to leave on this planet anymore :laughing:

Reply:
I'll keep this for a follow up
But `yy: { sectionType: 'year', contentType: 'digit', maxLength: 2 },` causes a bug as well... you can't enter a year below `9` because dayjs can't parse `9` for the token `YY` :facepalm:

Reply:
![image](https://github.com/user-attachments/assets/fb69a16f-0331-4790-9baf-d0f57a76e43a)

For me it's a bug on dayjs side because they clearly state it should have to digits, so `console.log(dayjs().set('year', 9).format('YY'))` should return `09`

Reply:
https://github.com/iamkun/dayjs/blob/45f88334d08f8278e51b8189671ddea4720c754e/src/index.js#L287

I think the fix is super simple here

Reply:
My comment was regarding the DateFns adapter, not dayjs

Reply:
https://github.com/iamkun/dayjs/pull/2847

Reply:
Oh sorry, in that case if you can extract it :pray: , I'll have a look at it

Reply:
It seems it was fixed after merging master in 🤔